### DISCUSSION TITLE
Clarify useDebounce does not support functions that take arguments?

### DISCUSSION THREAD

**Discussion by marckohlbrugge** [stimulus-use/stimulus-use#528] [General]
I ran into a problem when using `useDebounce`:

![telegram-cloud-photo-size-2-5239990659992119325-y](https://github.com/user-attachments/assets/fa5d7b10-829b-4f3e-81fa-be22c8165a06)

It somehow was using the input text of a field (`"amsterdam"` in this case) and treating it as an object. The same Stimulus controller worked fine without `useDebounce`.

After some debugging, I find out that `useDebounce` got confused when I passed in an argument to my debounced function. I think it has something to do with how `useDebounce` maintains `this`.

The following code change fixed my issue:

![telegram-cloud-photo-size-2-5239990659992119329-y](https://github.com/user-attachments/assets/af18871f-3314-43b8-9dff-68182f6687df)

TLDR: I think useDebounce does not support functions that take arguments. Therefore, I think this should be explained in the documentation, log a console error, and/or be updated to support this.

**Comment by adrienpoly** [comment]
This is indeed not expected nor documented. I ll see if there is a way to fix this.

I ll look also in my code as I have quite a few controller using `useDebounce` to see if I can reproduce it
